File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int res=0,count=0;
        for(int i=0;i<s.length();++i)
        {
            if(s.charAt(i)=='(')
                res=Math.max(res,++count);
           if(s.charAt(i)==')')
                count--;
        }
        return res;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution 
{
public:
    int maxDepth(string s) 
    {
        stack<char>c;
        int count=0;
        int res=0;
        
        for(char c:s)
        {
            if(c=='(')
            {
                count++;
                res=max(count,res);
            }
            else if(c==')')
            count--;
                
        }
        return res;
     }
};